compiler/src/middle_end/matchcomp.re

```diff
@@ -305,11 +305,29 @@ module MatchTreeCompiler = {
         | [] => []
         | _ => [(data_name, expr), ...binds]
         };
-      | TPatOr(_) => failwith("NYI: extract_bindings > TPatOr")
+      | TPatOr(left, right) =>
+        let left_binds = extract_bindings(left, expr);
+        let right_binds = extract_bindings(right, expr);
+        switch (left_binds, right_binds) {
+        | ([], []) => []
+        | (left_binds, []) => left_binds
+        | ([], right_binds) => right_binds
+        | (left_binds, right_binds) => List.concat([left_binds, right_binds])
+        };
       };
     };
     let res = extract_bindings(patt, expr);
-    /*Printf.eprintf "Bindings:\n%s\n" (Sexplib.Sexp.to_string_hum (sexp_of_list (sexp_of_pair sexp_of_string (fun x -> (sexp_of_string (Pretty.string_of_cexpr x)))) res));*/
+    /*
+       Printf.eprintf(
+         "Bindings:\n%s\n",
+         Sexplib.Sexp.to_string_hum(
+           sexp_of_list(
+             sexp_of_pair(Ident.sexp_of_t, sexp_of_comp_expression),
+             res,
+           ),
+         ),
+       );
+     */
```

Comment:
What's this? 👀

Reply:
A utility for printing bindings. It already existed in the code but it was in ocaml since comments don't get converted. I updated it to Reason and made it work with the current compiler.